bin/gravity_node/src/reth_cli.rs

```diff
@@ -223,16 +225,25 @@ impl<EthApi: RethEthCall> RethCli<EthApi> {
 
     pub async fn start_execution(&self) -> Result<(), String> {
         let mut start_ordered_block = self.provider.last_block_number().unwrap() + 1;
+        // Initialize current_epoch from block buffer manager
+        let buffer_epoch = get_block_buffer_manager().get_current_epoch().await;
+        self.current_epoch.store(buffer_epoch, Ordering::SeqCst);
+        info!("start_execution initialized with epoch {}", buffer_epoch);
+        
         loop {
+            let current_epoch = self.current_epoch.load(Ordering::SeqCst);
             // max executing block number
             let exec_blocks =
-                get_block_buffer_manager().get_ordered_blocks(start_ordered_block, None).await;
+                get_block_buffer_manager().get_ordered_blocks(start_ordered_block, None, current_epoch).await;
             if let Err(e) = exec_blocks {
                 let from = start_ordered_block;
                 if e.to_string().contains("Buffer is in epoch change") {
```

Comment:
use one meaningful error type?

Reply:
Let's mention a separate PR later to modify this part